src/battle_setup.c

```diff
@@ -1655,30 +1657,44 @@ static void SetRematchIdForTrainer(const struct RematchTrainer *table, u32 table
     gSaveBlock1Ptr->trainerRematches[tableId] = i;
 }
 
+static bool32 DoesCurrentMapMatchRematchTrainerMap(s32 i, const struct RematchTrainer *table, u16 mapGroup, u16 mapNum)
+{
+    return table[i].mapGroup == mapGroup && table[i].mapNum == mapNum;
+}
+
+bool32 TrainerIsMatchCallRegistered(s32 i)
+{
+    return FlagGet(FLAG_MATCH_CALL_REGISTERED + i);
+}
+
 static bool32 UpdateRandomTrainerRematches(const struct RematchTrainer *table, u16 mapGroup, u16 mapNum)
 {
     s32 i;
-    bool32 ret = FALSE;
+
+#if I_VS_SEEKER_CHARGING != 0
+    if (CheckBagHasItem(ITEM_VS_SEEKER, 1) == TRUE)
+        return FALSE;
+#endif
```

Comment:
```suggestion
    if (I_VS_SEEKER_CHARGING != 0 && CheckBagHasItem(ITEM_VS_SEEKER, 1) == TRUE)
        return FALSE;
```
Perhaps the `== TRUE` isn't necessary either

Reply:
Fixed in https://github.com/PokemonSanFran/pokeemerald-expansion/commit/e4fe0e19eff65df5e93af055551d598e7ff4fcb8